src/CabanaPD_Boundary.hpp

```diff
@@ -279,6 +267,22 @@ auto createBoundaryCondition( BCTag, ExecSpace exec_space, Particles particles,
     return BoundaryCondition<bc_index_type, BCTag>( value, bc_indices );
 }
 
+// FIXME: relatively large initial guess for allocation.
+template <class UserFunctor, class BoundaryType, class ExecSpace,
+          class Particles>
+auto createBoundaryCondition( UserFunctor user_functor, ExecSpace exec_space,
+                              Particles particles,
+                              std::vector<BoundaryType> planes,
+                              const double initial_guess = 0.5 )
```

Comment:
@sreeve : What is "initial_guess = 0.5" for?

Reply:
We have to allocate before we know how many boundary particles there are